RoaringBitmap/src/main/java/org/roaringbitmap/art/Art.java

```diff
@@ -128,7 +128,7 @@ protected Toolkit removeSpecifyKey(Node node, byte[] key, int dep) {
     }
     if (node.prefixLength > 0) {
       int mismatchIndex = ArraysShim.mismatch(node.prefix, 0,
-          node.prefixLength, key, dep, node.prefixLength);
+          node.prefixLength, key, dep, dep + node.prefixLength);
```

Comment:
It would be better to add a check to make sure `dep + node.prefixLength` not larger than key.length.

Reply:
Fixed, i reused the `commonPrefixLength` method since it has all the relevant logic. PTAL.